.travis.yml

```diff
@@ -1,31 +1,43 @@
-language: haskell
-
-matrix:
-  include:
-    - os: linux
-    - os: osx
+cache: apt
+env:
+ - GHCVER=7.8.4
+ - GHCVER=7.8.3
+ - GHCVER=7.6.3
 
+before_install:
+ - sudo add-apt-repository -y ppa:hvr/ghc
+ - sudo apt-get update
+ - sudo apt-get install zookeeperd 
+ - sudo apt-get install cabal-install-1.20 ghc-$GHCVER
+ - export PATH=/opt/ghc/$GHCVER/bin:$PATH
+ - |
+   if [ $GHCVER = "head" ] || [ ${GHCVER%.*} = "7.8" ]; then
+     sudo apt-get install happy-1.19.4 alex-3.1.3
+     export PATH=/opt/alex/3.1.3/bin:/opt/happy/1.19.4/bin:$PATH
+   else
+     sudo apt-get install happy alex
+   fi
 install:
-    - cabal update
+ - cabal-1.20 update
 
 script:
   - cd test-sandbox
-  - cabal install --only-dependencies --enable-tests
-  - "cabal configure --enable-tests --ghc-options=-Werror && cabal build && cabal test "
-  - "cabal sdist && cabal install dist/test-sandbox-*.tar.gz"
+  - cabal-1.20 install --only-dependencies --enable-tests
+  - "cabal-1.20 configure --enable-tests --ghc-options=-Werror && cabal-1.20 build && cabal-1.20 test "
+  - "cabal-1.20 sdist && cabal-1.20 install dist/test-sandbox-*.tar.gz"
```

Comment:
This script will not run the test suite within the set of files that's contained in the release tarballs. As a result, you won't notice if the test suite ever depends on a file that's not listed in `other-sources` and thus not included in the release archive.